File: Assignment7.ml
let read_int_list n =
  let rec aux i acc =
    if i = 0 then acc
    else (
      print_string "Enter number: ";
      let x = read_int () in
      aux (i - 1) (x :: acc)
    )
  in aux n []

let mean lst =
  let sum = List.fold_left (+.) 0.0 (List.map float_of_int lst) in
  sum /. float_of_int (List.length lst)

let median lst =
  let sorted = List.sort compare lst in
  let len = List.length sorted in
  if len mod 2 = 0 then
    let a = List.nth sorted (len / 2 - 1) in
    let b = List.nth sorted (len / 2) in
    (float_of_int (a + b)) /. 2.0
  else
    float_of_int (List.nth sorted (len / 2))

let mode lst =
  let count_map = List.fold_left (fun acc x ->
    let count = try List.assoc x acc + 1 with Not_found -> 1 in
    (x, count) :: List.remove_assoc x acc
  ) [] lst in
  let max_freq = List.fold_left (fun acc (_, c) -> max acc c) 0 count_map in
  List.filter (fun (_, c) -> c = max_freq) count_map
  |> List.map fst

let () =
  print_string "Enter number of elements: ";
  let n = read_int () in
  let data = read_int_list n in
  Printf.printf "Mean: %.2f\n" (mean data);
  Printf.printf "Median: %.2f\n" (median data);
  Printf.printf "Mode: ";
  List.iter (Printf.printf "%d ") (mode data);
  print_newline ()

File: Assignment7.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

double calculate_mean(int arr[], int n) {
    double sum = 0;
    for (int i = 0; i < n; i++) sum += arr[i];
    return sum / n;
}

double calculate_median(int arr[], int n) {
    qsort(arr, n, sizeof(int), compare);
    if (n % 2 == 0)
        return (arr[n / 2 - 1] + arr[n / 2]) / 2.0;
    else
        return arr[n / 2];
}

void calculate_mode(int arr[], int n) {
    int max_count = 0;
    printf("Mode: ");
    for (int i = 0; i < n; i++) {
        int count = 0;
        for (int j = 0; j < n; j++)
            if (arr[j] == arr[i]) count++;
        if (count > max_count) max_count = count;
    }

    int printed[201] = {0};
    for (int i = 0; i < n; i++) {
        int count = 0;
        if (printed[arr[i] + 100]) continue;
        for (int j = 0; j < n; j++)
            if (arr[j] == arr[i]) count++;
        if (count == max_count) {
            printf("%d ", arr[i]);
            printed[arr[i] + 100] = 1;
        }
    }
    printf("\n");
}

int main() {
    int arr[100], n;
    printf("Enter number of elements: ");
    scanf("%d", &n);

    printf("Enter %d integers:\n", n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    printf("Mean: %.2f\n", calculate_mean(arr, n));
    printf("Median: %.2f\n", calculate_median(arr, n));
    calculate_mode(arr, n);

    return 0;
}
